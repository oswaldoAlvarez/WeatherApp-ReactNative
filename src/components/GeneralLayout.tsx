import React, { FunctionComponent } from 'react';
import { SafeAreaView, StatusBar } from 'react-native';
import { appTheme } from '../theme/appTheme';
import { NavigationContainer } from '@react-navigation/native';
import { appStateProps } from '../interfaces/generalInterfaces';
import { Navigator } from '../navigator/Navigator';

export const GeneralLayout: FunctionComponent = () => {
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <StatusBar backgroundColor="rgba(52, 52, 52, 0.8)" barStyle="light-content" />
      <NavigationContainer theme={appTheme}>
        <AppState>
          <Navigator />
        </AppState>
      </NavigationContainer>
    </SafeAreaView>
  );
};

const AppState: FunctionComponent<appStateProps> = ({ children }) => {
  return <>{children}</>;
};
