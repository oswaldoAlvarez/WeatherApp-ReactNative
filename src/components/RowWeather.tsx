import React, { FunctionComponent } from 'react';
import { View, Text, StyleSheet } from 'react-native';

interface Props {
  weather: any;
  descripcion: string;
  signo?: string;
}

export const RowWeather: FunctionComponent<Props> = ({ weather, descripcion, signo }) => {
  return (
    <View style={styles.rowWeather}>
      <Text style={styles.descriptionWeather}>{descripcion}</Text>
      <Text style={styles.weather}>
        {weather}
        {signo}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  rowWeather: {
    alignItems: 'center',
    marginTop: 10,
    marginBottom: 10,
    marginRight: 20,
    padding: 10,
  },
  descriptionWeather: {
    fontSize: 20,
    fontWeight: 'bold',
    fontFamily: 'sans-serif-medium',
    color: 'white',
  },
  weather: {
    fontSize: 17,
    color: 'white',
  },
});
