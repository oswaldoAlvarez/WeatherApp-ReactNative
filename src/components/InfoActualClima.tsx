import React, { FunctionComponent } from 'react';
import { View, Text, ScrollView, StyleSheet } from 'react-native';
import { RowWeather } from './RowWeather';

interface Props {
  infoTitulo: string;
  weather: any;
}

export const InfoActualClima: FunctionComponent<Props> = ({ infoTitulo, weather }) => {
  if (!weather) return <Text style={[styles.loading, styles.marginBottom30, styles.marginTop50]}>Cargando...</Text>;
  return (
    <>
      {weather ? (
        <>
          <Text style={styles.infoClima}>{infoTitulo}</Text>
          <ScrollView horizontal={true}>
            <View style={styles.responseContainer}>
              <RowWeather weather={weather.name} descripcion="Ciudad:" />
              <RowWeather weather={weather.sys.country} descripcion="País:" />
              <RowWeather weather={weather.main.temp} descripcion="Temperatura Actual:" signo="°C" />
              <RowWeather weather={weather.main.temp_min} descripcion="Temperatura mínima de hoy:" signo="°C" />
              <RowWeather weather={weather.main.temp_max} descripcion="Temperatura máxima de hoy:" signo="°C" />
              <RowWeather weather={weather.main.feels_like} descripcion="Sensación térmica:" signo="°C" />
              <RowWeather weather={weather.clouds.all} descripcion="Nubocidad:" signo="%" />
              <RowWeather weather={weather.wind.speed} descripcion="Viento:" signo="K/H" />
              <RowWeather weather={weather.main.humidity} descripcion="Humedad:" signo="%" />
              <RowWeather weather={weather.weather[0].main} descripcion="Clima:" />
              <RowWeather weather={weather.weather[0].description} descripcion="Descripción del Clima:" />
            </View>
          </ScrollView>
        </>
      ) : (
        <Text style={[styles.loading, styles.marginBottom30]}>Ha ocurrido un error con el endpoint o la ciudad!</Text>
      )}
    </>
  );
};

const styles = StyleSheet.create({
  infoClima: {
    fontSize: 16,
    fontWeight: 'bold',
    fontFamily: 'sans-serif-condensed',
    color: 'white',
    textDecorationLine: 'underline',
  },
  responseContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    paddingBottom: 10,
  },
  loading: {
    fontSize: 30,
    textAlign: 'center',
    color: 'white',
  },
  marginTop50: {
    marginTop: 50,
  },
  marginBottom30: {
    marginBottom: 30,
  },
});
