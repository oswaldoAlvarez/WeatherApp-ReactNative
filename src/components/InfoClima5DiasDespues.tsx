import React, { FunctionComponent } from 'react';
import { View, Text, ScrollView, StyleSheet } from 'react-native';
import { RowWeather } from './RowWeather';

interface Props {
  weather: any;
  weatherWeek: any;
}

export const InfoClima5DiasDespues: FunctionComponent<Props> = ({ weather, weatherWeek }) => {
  return (
    <>
      {weatherWeek ? (
        <>
          <Text style={styles.infoClima}>
            Temperatura de ({weather.name}, {weather.sys.country}) en los próximos 5 días:
          </Text>
          <ScrollView horizontal={true}>
            <View style={styles.responseContainer}>
              <RowWeather weather={weatherWeek.list[0].main.temp} descripcion="Primer día después:" signo="°C" />
              <RowWeather weather={weatherWeek.list[1].main.temp} descripcion="Segundo día después:" signo="°C" />
              <RowWeather weather={weatherWeek.list[2].main.temp} descripcion="Tercer día después:" signo="°C" />
              <RowWeather weather={weatherWeek.list[3].main.temp} descripcion="Cuarto día después:" signo="°C" />
              <RowWeather weather={weatherWeek.list[4].main.temp} descripcion="Quinto día después:" signo="°C" />
            </View>
          </ScrollView>
        </>
      ) : (
        <Text>Ha ocurrido un error con el endpoint o la ciudad!</Text>
      )}
    </>
  );
};

const styles = StyleSheet.create({
  responseContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  infoClima: {
    fontSize: 16,
    fontWeight: 'bold',
    fontFamily: 'sans-serif-condensed',
    color: 'white',
    textDecorationLine: 'underline'
  },
});
