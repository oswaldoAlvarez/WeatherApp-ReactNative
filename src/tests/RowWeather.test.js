import React from 'react';
import renderer from 'react-test-renderer';
import { RowWeather } from '../components/RowWeather';

test('Probando funciones del RowWeather', () => {
  const tree = renderer.create(<RowWeather />).toJSON();
  expect(tree).toMatchSnapshot();
});
