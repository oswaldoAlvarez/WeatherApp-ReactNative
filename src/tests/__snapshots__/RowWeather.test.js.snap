// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando funciones del RowWeather 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginBottom": 10,
      "marginRight": 20,
      "marginTop": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "sans-serif-medium",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 17,
      }
    }
  />
</View>
`;
